Description: Support either ImageMagick 6 or 7.
Author: Doug Torrance <dtorrance@piedmont.edu>
	Roman Dobosz <gryf73@gmail.com>
Bug-Debian: https://bugs.debian.org/929825
Forwarded: https://groups.google.com/g/wmaker-dev/c/tpWR2oVpXHM/m/tkgWSHg1BwAJ
	   https://groups.google.com/g/wmaker-dev/c/a0PyeSLDhdE/m/BGuWFGdRBwAJ
	   https://groups.google.com/g/wmaker-dev/c/h5c9TTZ5BnA/m/o3hzrmapBwAJ
Applied-Upstream: 0.95.10,
 https://repo.or.cz/wmaker-crm.git/commitdiff/dfa92906
 https://repo.or.cz/wmaker-crm.git/commitdiff/3022edd0
 https://repo.or.cz/wmaker-crm.git/commitdiff/230a501d
Last-Update: 2021-10-08

--- a/m4/wm_imgfmt_check.m4
+++ b/m4/wm_imgfmt_check.m4
@@ -313,8 +313,12 @@
               dnl The library was found, check if header is available and compiles
               wm_save_CFLAGS="$CFLAGS"
               AS_IF([wm_fn_lib_try_compile "MagickWand/MagickWand.h" "MagickWand *wand;" "wand = NewMagickWand()" "$wm_cv_libchk_magick_cflags"],
-                  [wm_cv_libchk_magick="$wm_cv_libchk_magick_cflags % $wm_cv_libchk_magick_libs"],
-                  [AC_MSG_ERROR([found MagickWand library but could not compile its header])])
+                  [wm_cv_libchk_magick="$wm_cv_libchk_magick_cflags % $wm_cv_libchk_magick_libs"
+		   wm_cv_libchk_magick_version=7],
+		  [wm_fn_lib_try_compile "wand/magick_wand.h" "MagickWand *wand;" "wand = NewMagickWand()" "$wm_cv_libchk_magick_cflags"],
+                  [wm_cv_libchk_magick="$wm_cv_libchk_magick_cflags % $wm_cv_libchk_magick_libs"
+		   wm_cv_libchk_magick_version=6],
+		  [AC_MSG_ERROR([found MagickWand library but could not compile its header])])
               CFLAGS="$wm_save_CFLAGS"])dnl
          ])
      AS_IF([test "x$wm_cv_libchk_magick" = "xno"],
@@ -323,7 +327,7 @@
          [supported_gfx="$supported_gfx Magick"
           MAGICKFLAGS=`echo "$wm_cv_libchk_magick" | sed -e 's, *%.*$,,' `
            MAGICKLIBS=`echo "$wm_cv_libchk_magick" | sed -e 's,^.*% *,,' `
-          AC_DEFINE([USE_MAGICK], [1],
+          AC_DEFINE_UNQUOTED([USE_MAGICK], [$wm_cv_libchk_magick_version],
               [defined when MagickWand library with header was found])])
      ])
 AM_CONDITIONAL([USE_MAGICK], [test "x$enable_magick" != "xno"])dnl
--- a/wrlib/load_magick.c
+++ b/wrlib/load_magick.c
@@ -22,7 +22,13 @@
 
 #include "config.h"
 
+#ifdef USE_MAGICK
+#if USE_MAGICK < 7
+#include <wand/magick_wand.h>
+#else
 #include <MagickWand/MagickWand.h>
+#endif
+#endif
 
 #include "wraster.h"
 #include "imgformat.h"
